lib/matplotlib/figure.py

```diff
@@ -1377,14 +1380,26 @@ def add_subplot(self, *args, **kwargs):
                 raise TypeError('Positional arguments are not a valid '
                                 'position specification.')
         elif nargs == 3:
-            for arg in args:
+            newarg = [None] * 3
+            message = ("Passing non-integers as three-element "
+                       "position specification is deprecated since "
+                       "%(since)s and will be removed %(removal)s.")
+            for nn, arg in enumerate(args[:2]):
                 if not isinstance(arg, Integral):
-                    cbook.warn_deprecated(
-                        "3.3",
-                        message="Passing non-integers as three-element "
-                                "position specification is deprecated since "
-                                "%(since)s and will be removed %(removal)s.")
-            args = tuple(map(int, args))
+                    cbook.warn_deprecated("3.3", message=message)
+                newarg[nn] = int(arg)
+            args2 = args[2]
```

Comment:
Unused... I assume this should replace all `args[2]` below, though.

Reply:
I really shouldn't do this stuff while also watching the kids...